cylc/rose/platform_utils.py

```diff
@@ -47,15 +49,30 @@ def get_platform_from_task_def(
     config = WorkflowConfig(flow, flow_file, Values())
     # Get entire task spec to allow Cylc 7 platform from host guessing.
     task_spec = config.pcfg.get(['runtime', task])
+    # check for subshell and evaluate
+    if 'platform' in task_spec and (
+        task_spec['platform'] is not None) and (
+            is_platform_definition_subshell(task_spec['platform'])):
+        task_spec['platform'] = eval_subshell(task_spec['platform'])
+    elif 'remote' in task_spec and (
+        'host' in task_spec['remote']) and (
+        task_spec['remote']['host'] is not None) and (
+            bool(HOST_REC_COMMAND.match(task_spec['remote']['host']))):
+        task_spec['remote']['host'] = eval_subshell(
+            task_spec['remote']['host'])
     platform = get_platform(task_spec)
-    if platform is None:
-        raise PlatformLookupError(
-            'Platform lookup failed; platform is a subshell to be evaluated: '
-            f' Task: {task}, platform: {task_spec["platform"]}.'
-        )
     return platform
 
 
+def eval_subshell(platform):
+    """Evaluates platforms/hosts defined as subshell"""
+    match = HOST_REC_COMMAND.match(platform)
+    output = subprocess.run(
+        ['bash', '-c', f" {match[2]}"], capture_output=True
+    )
+    return output.stdout.strip().decode()
```

Comment:
Why the f-string? 

Also, `text=True` avoids the need to decode later
```suggestion
    output = subprocess.run(
        ['bash', '-c', match[2]], capture_output=True, text=True
    )
    return output.stdout.strip()
```